core/src/main/java/com/sap/ai/sdk/core/AiCoreDeployment.java

```diff
@@ -0,0 +1,224 @@
+    this(service, deploymentId, "default");
+  }
+
+  /**
+   * Create a new instance of the AI Core service with a deployment.
+   *
+   * @param service The AI Core service.
+   * @param modelName The model name.
+   * @return A new instance of the AI Core service.
+   */
+  @Nonnull
+  public static AiCoreDeployment forModelName(
+      @Nonnull final AiCoreService service, @Nonnull final String modelName) {
+    final Predicate<AiDeployment> p = deployment -> isDeploymentOfModel(modelName, deployment);
+    return new AiCoreDeployment(service, obj -> obj.getDeploymentId(service.client(), p));
+  }
+
+  /**
+   * Create a new instance of the AI Core service with a deployment.
+   *
+   * @param service The AI Core service.
+   * @param scenarioId The scenario id.
+   * @return A new instance of the AI Core service.
+   */
+  @Nonnull
+  public static AiCoreDeployment forScenarioId(
+      @Nonnull final AiCoreService service, @Nonnull final String scenarioId) {
+    final Predicate<AiDeployment> p = deployment -> scenarioId.equals(deployment.getScenarioId());
+    return new AiCoreDeployment(service, obj -> obj.getDeploymentId(service.client(), p));
+  }
+
+  /**
+   * Create a new instance of the AI Core service with a deployment.
+   *
+   * @param service The AI Core service.
+   * @param deploymentId The deployment id.
+   * @return A new instance of the AI Core service.
+   */
+  @Nonnull
+  public static AiCoreDeployment forDeploymentId(
+      @Nonnull final AiCoreService service, @Nonnull final String deploymentId) {
+    return new AiCoreDeployment(service, obj -> deploymentId);
+  }
+
+  @Nonnull
+  @Override
+  public Destination destination() {
+    final var dest = service.baseDestinationHandler.apply(service);
+    final var builder = service.builderHandler.apply(service, dest);
+    destinationSetUrl(builder, dest);
+    destinationSetHeaders(builder, dest);
+    return builder.build();
+  }
+
+  @Nonnull
+  @Override
+  public ApiClient client() {
+    final var destination = destination();
+    return service.clientHandler.apply(service, destination);
+  }
+
+  /**
+   * Update and set the URL for the destination.
+   *
+   * @param builder The destination builder.
+   * @param dest The original destination reference.
+   */
+  protected void destinationSetUrl(
+      @Nonnull final DefaultHttpDestination.Builder builder, @Nonnull final Destination dest) {
+    String uri = dest.get(DestinationProperty.URI).get();
+    if (!uri.endsWith("/")) {
+      uri = uri + "/";
+    }
+    builder.uri(uri + "v2/inference/deployments/%s/".formatted(getDeploymentId()));
+  }
+
+  /**
+   * Update and set the default request headers for the destination.
+   *
+   * @param builder The destination builder.
+   * @param dest The original destination reference.
+   */
+  protected void destinationSetHeaders(
+      @Nonnull final DefaultHttpDestination.Builder builder, @Nonnull final Destination dest) {
```

Comment:
```suggestion
   */
  protected void destinationSetHeaders(@Nonnull final DefaultHttpDestination.Builder builder) {
```

Parameter dest unused